Staging (#340)
macrocosm-os/folding: documentation/get_started.md

# Get Started with SN25 

## Installation

Core software requirements include:
```
conda
poetry 
```

Core hard/firmware requirements include: 
```
linux
CUDA 

> RXT 4090

Reviewer: ```suggestion
RTX 4090
```

```

> To start, simply clone this repository and run the [install.sh](./install.sh) script. Below are all the steps needed to ensure that your machine is running properly.

Reviewer: ```suggestion
To start, simply clone this repository and run the [install.sh](./install.sh) script. Below are steps to ensure your machine is running properly.
```

Firstly, you must install conda: 
```bash 
mkdir -p ~/miniconda3
wget https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh -O ~/miniconda3/miniconda.sh
bash ~/miniconda3/miniconda.sh -b -u -p ~/miniconda3
rm ~/miniconda3/miniconda.sh

source ~/miniconda3/bin/activate
conda init --all
```

Install wandb (optional as a miner): 
```bash
pip install wandb 
wandb login
```

We use a combination of `conda` and `poetry` to manage our environments. It is very important to create the environment with python 3.11 as this is necesssary for `bittensor` and `openmm`
```bash 
git clone https://github.com/macrocosm-os/folding.git
cd folding

conda create --name folding python=3.11
bash install.sh
```

Bash install will use poetry to build the environment correctly. 

## Launch Commands
### Validator

> SN25 runs off DigitalOcean S3 data buckets for data transfer. Therefore, the following environment variables need to be set in your system or application environment (`.env` file):

Reviewer: ```suggestion
SN25 uses DigitalOcean S3 data buckets for data transfer. Therefore, the following environment variables need to be set in your system or application environment (`.env` file):
```
